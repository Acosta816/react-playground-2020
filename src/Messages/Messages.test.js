import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Messages from './Messages';

describe('TestSuite: Messages Component', ()=> {

    it('SMOKE: should render without crashing', ()=> {
        const div = document.createElement('div');
    
        ReactDOM.render(<Messages />, div);
    
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('SNAPSHOT#01: renders the UI as expected compared to the saved shot', ()=> {
    
        //Render the component and create a human-readable JSON file.
        const tree = renderer.create( <Messages name="Messages" unread={4} /> ).toJSON();
    
        //Compare rendered component to the saved version of that component.
        expect(tree).toMatchSnapshot();
    })
    
    it('SNAPSHOT#02: renders the UI as expected with no unreads', () => {
        const tree = renderer
          .create(<Messages name="Messages" unread={0}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});

