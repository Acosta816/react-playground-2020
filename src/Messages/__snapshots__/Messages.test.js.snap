// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuite: Messages Component SNAPSHOT#01: renders the UI as expected compared to the saved shot 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`TestSuite: Messages Component SNAPSHOT#02: renders the UI as expected with no unreads 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
